lib/rspec/core/example_group.rb

```diff
@@ -340,6 +340,16 @@ def self.include_examples(name, *args, &block)
         find_and_eval_shared("examples", name, caller.first, *args, &block)
       end
 
+      # Adds an example to the example group
+      def self.add_example example
+        examples << example
+      end
+
+      # Removes an example from the example group
+      def self.remove_example example
+        examples.delete example
+      end
```

Comment:
I've updated the PR to reset the memoized values for:

```
@descendant_filtered_examples = nil
@_descendants = nil 
@parent_groups = nil
@declaration_line_numbers = nil
```

I also added a test to verify the object ids returned are different. Let me know if any further changes are required.